toolchain/semantics/semantics_node.h

```diff
@@ -68,6 +68,21 @@ struct SemanticsCrossReferenceIRId : public IndexBase {
   }
 };
 
+// The ID of a boolean value.
+struct SemanticsBoolLiteralId : public IndexBase {
```

Comment:
Maybe drop the "Id" from the name and comment, since this is the literal value. No concern sticking with IndexBase